src/browser-provider/index.js

```diff
@@ -0,0 +1,53 @@
+import { assignIn } from 'lodash';
+import { PathBrowserProvider, LocalBrowserProvider, RemoteBrowserProvider } from './built-in';
+
+
+export var BUILT_IN_PROVIDERS = {
+    'local':  LocalBrowserProvider,
+    'path':   PathBrowserProvider,
+    'remote': RemoteBrowserProvider
+};
+
+
+export default class BrowserProvider {
+    constructor (providerName) {
+        var provider = BUILT_IN_PROVIDERS[providerName];
+
+        if (!provider)
+            provider = require(`testcafe-provider-${providerName}`);
```

Comment:
Also, it's currently missing logic for handling unknown provider: see https://github.com/AndreyBelym/testcafe/blob/28b0b77f72f7973a1dbc9c9939ae8a4d0d6d1042/src/runner/bootstrapper.js#L111 for example